tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionDebugging.cs

```diff
@@ -77,9 +77,22 @@ private static void InitSnapshotsSink()
                 uri => uri);
             var discoveryService = tracer.TracerManager.DiscoveryService;
             var gitMetadataTagsProvider = tracer.TracerManager.GitMetadataTagsProvider;
-            var batchApi = AgentBatchUploadApi.Create(apiFactory, discoveryService, gitMetadataTagsProvider);
+
+            var snapshotBatchUploadApi = AgentBatchUploadApi.Create(apiFactory, discoveryService, gitMetadataTagsProvider, false);
+            var snapshotBatchUploader = BatchUploader.Create(snapshotBatchUploadApi);
+
+            var diagnosticsBatchUploadApi = AgentBatchUploadApi.Create(apiFactory, discoveryService, gitMetadataTagsProvider, true);
+            var diagnosticsBatchUploader = BatchUploader.Create(diagnosticsBatchUploadApi);
```

Comment:
In Exception Debugging (aka Exception Replay), the diagnostics batch uploader is non-relevant (that's why I have the `NopProbeStatusSink` below), can you spare it's creation?

Reply:
Got it! Introduced `IBatchUploader` and `NonBatchUploader` here 457f68b62abcaf7d5847ae627b451bc8bf02627f